### ISSUE TITLE
AttributeError using mkdocs serve

### ISSUE THREAD

**Issue by sirchampsblanc** [mkdocs/mkdocs#2446]
I updated my current virtual env to use mkdocs 1.2 (i'm testing the new analytics feature since i can't generate a UA-XXXXXX to use with older versions) and i get an error at the end of the building while serving (mkdocs build works fine):

```
File ".venv/lib/python3.6/site-packages/mkdocs_monorepo_plugin/plugin.py", line 67, in on_serve
    buildfunc = list(server.watcher._tasks.values())[0]['func']
AttributeError: 'LiveReloadServer' object has no attribute 'watcher'
```

Any ideas why? Thanks!

(Let me know if i need to add more details or library versions)

**Comment by waylan** [comment]
In 1.2 we completely replaced the server with a new one. It would appear that the plugin you are using (`mkdocs_monorepo_plugin`) is doing something unusual with the file watcher. Specifically, it is looking up attributes that we don't document (in fact, they are designated as private with the `underscore`: `_tasks`). Our new implementation no longer has those private attributes. Therefore, that plugin won't work until it is updated to work with the new file watcher. As this resulted from accessing attributes which were clearly labeled as private, the issue lies with the plugin. I would suggest reporting this issue to the plugin developer.

**Comment by oprypin** [comment]
FYI for anyone looking to fix their plugin, here are some examples:
* A "modern" fix that is compatible only with MkDocs 1.1.1 and the new 1.2:
https://github.com/mkdocstrings/mkdocstrings/commit/e8d657352aec2e519971ef70903c0cdb72bb4418
* A fix including more legacy versions:
https://github.com/fralau/mkdocs_macros_plugin/commit/4bf847b3460471ee9cf2559cd5562954f378fe3b
